pecos/xmc/base.py

```diff
@@ -1954,3 +1955,23 @@ def pifa_lf_convex_combine(Y, X, Z, alpha=0.5, threads=-1, normalized_Y=True):
             label_embedding = alpha[:, None] * pifa + alpha_complementary[:, None] * Z
 
         return label_embedding
+
+    @staticmethod
+    def pii(Y, X=None):
+        """Create PII label embedding by normalize along each label.
+
+        Args:
+            Y (smat.spmatrix): label matrix (num_samples x num_labels).
+            X: Unused variable to match interface.
+        Returns:
+            label_embedding (csr_matrix): pii embeddings. (num_labels x num_samples)
+        """
+        del X  # delete unused
```

Comment:
we should not delete the input parameter X here. the caller might still need X for other purpose.

Reply:
Delete reference inside function will not affect anything outside, because the reference is a new one for the function's scope.

Deleting is only one of the methods to address non-used variables in function. However, it does not result in any type check issues for our current settings. I'll remove this line for now for simplicity although I do think we should use other methods to address it is a non-used variable.

Reply:
Give up unifying factory method interface. Let `pii` has its own interface while making branch in the factory method when creating.